Spaghettification.ipynb

```diff
@@ -0,0 +1,2375 @@
qYUr9cuXKKiIjIs6VHdna24uLiFBUV5fDcAQMGuLVmQkKCXcBZrVo1PfHEE27VAmA9b75OFdRvv/2mX375xW7stttu89h6AMznC8FBv3797ALOefPmadKkSQoMvP4/25KTk/Xtt9/ajXXt2vW65wHwHr58L3Xt05vlypXTnXfe6XY9AN7H2++lWrVqpYMHD175/P333ysnJ8fprXOTk5O1du1au7GWLVs6dS4AwBpsUQv4iLi4OMPx2rVrm7ZGnTp1XFobAK7m7dcpm83m1ppJSUl5nkAICAjQgw8+6FY9ANYorODgWpeDA2f87W9/s/sG4YkTJ/TWW285de4rr7xit62an5+fHnroIafOBeAdvP1eypHs7Gx9/vnndmMPPPCAgoKC3K4JwPt4+71UTEyM3ee4uDi99957Tq8/fvx4paen51sTAOBdCDgBH3H8+HHD8apVq5q2RpUqVVxaGwCu5u3XqWeeeUYvvviiEhMTnV5v9+7d6tSpkw4dOmQ3PnToUDVq1MjpOgCs5wvBQVBQkMaPH2839sorr+ibb77J97yZM2dqxowZdmN9+/ZVs2bNnFoXgHfw9nspR5YtW6ZTp07ZjQ0aNMjtegC8k7ffS/Xv3z/Pe39HjBiRZ4cLI9OmTdO0adPsxqKiokx5tygAwHMIOAEfcebMGcNxM98HcO2N4PXWBoCreft1KiUlRa+//rpq1KihmJgYvfvuu1q/fn2edwinpKRo2bJlGjJkiG644Qbt3r3bbr5x48aaPHmy+18EAEv4SnAwYMAAdenS5crnS5cu6d5779XQoUO1ceNG5ebmSpJycnL0yy+/6J577tGTTz5p95R65cqV9c4777j5VQCwirffSzny8ccf232Oiopy+C49AL7L2++lwsLC9P/+3/+zG8vKylLfvn31wAMPaNWqVcrMzLwyl5KSosWLF6t79+4aMWKE3Xl+fn6aMmWK09vbAgCswTs4AR9x7XtSLgsPDzdtjdKlS7u0NgBczVeuU9nZ2frxxx/1448/XhkLCgpS6dKllZGRoQsXLjg8NzIyUsuWLTPcOgmAd/OV4MDf319ffPGFOnbsqL1790r6a4vtjz76SB999JECAwMVHh6u5OTkK2Hn1cLDw/X111+rWrVq7n0RACzjK/dSV0tOTtaSJUvsxq7d2h9A0eAL91JPPPGEduzYYbezhc1m0/z58zV//nz5+fmpTJkyysnJyfODrlebNGkS7xEGAB/AE5yAj7j6nUpXK1WqlGlrOKrlaG0AuJovX6eysrJ05swZh+Gmn5+fHn74YW3dulW1atUq0FoArOFLwUHFihW1du1ade3aNc9cdna2zp49axhuNmjQQOvWrdOtt97q0noAvIMv3kt98cUXunjx4pXP/v7+vP8XKKJ85V5q+vTpevPNNxUcHJxnzmazKTk52WG4GR4erjlz5mjMmDFOrwcAsA4BJ+AjsrKyDMcDA817ENtRLUdrA8DVvP06NWLECL366qvq2LGjwsLCnFqvWrVqGj58uHbv3q3Zs2c7fR4A7+NrwUHFihX1008/ae7cubrhhhvyPTYqKkozZszQzp071bRpU5fXAuAdvP1eysgnn3xi97lr166qUaOGW7UAeDdfuZfy8/PTc889pz179mjkyJGqXLnydc+pXbu2xo0bp3379vEUOgD4ELaoBXxEdna24biZ7wNw9I/dS5cumbbG1caNG6dx48Z5pDaAwuft16mWLVuqZcuWkv76yd2DBw/q8OHDOn78uM6dO6fMzEwFBwcrIiJCFSpUUOvWrXlaEyhCfDE48PPz04ABAzRgwADFx8frjz/+0NGjR5WZmamIiAhVqlRJN954I9cqoIjw9nspI7/99ltB2gHgQ3ztXqpu3bqaMmWKpkyZokOHDmnLli06ffq0zp8/Lz8/P5UtW1aVK1dWmzZt+MEMAPBRBJyAj3B0k5eTk2PaGo7+QV2iRAnT1gBQdPnSdcrPz08NGjRQgwYNzGgLgA/wxeDganXq1FGdOnUKXAeA9/KleykAxY8v30tFRkYqMjKyQDUAAN6HLWoBHxEUFGQ47ugG0x2OajlaGwCuxnUKgDcjOADg7biXAuDNuJcCAHgbAk7AR4SGhhqOp6enm7aGo1ohISGmrQGg6OI6BcCbERwA8HbcSwHwZtxLAQC8DQEn4CMiIiIMx1NSUkxbIzU11XC8XLlypq0BoOjiOgXAmxEcAPB23EsB8GbcSwEAvA0BJ+AjypcvbzielJRk2hqnT592aW0AuBrXKQDejOAAgLfjXgqAN+NeCgDgbQg4AR9RvXp1w/GEhATT1khMTHRpbQC4GtcpAN6M4ACAt+NeCoA3414KAOBtCDgBH1G3bl3D8fj4eNPWiIuLc2ltALga1ykA3ozgAIC3414KgDfjXgoA4G0IOAEf0aBBA8PxgwcPmlL/3LlzOnfuXJ7xgIAA/rELwClcpwB4M4IDAN6OeykA3ox7KQCAtyHgBHxE69atDcfPnz9vys3kli1bDMcbN26skiVLFrg+gKKP6xQAb0ZwAMDbcS8FwJtxLwUA8DYEnICPqFChgmrXrm049/vvvxe4/h9//GE4Hh0dXeDaAIoHrlMAvBnBAQBvx70UAG/GvRQAwNsQcAI+pEuXLobjK1euLHBtRzW6du1a4NoAig+uUwC8FcEBAF/AvRQAb8W9FADA2xBwAj4kJibGcHzJkiXKzc11u+65c+e0du3aPON+fn7q0aOH23UBFD9cpwB4M4IDAN6OeykA3ox7KQCANyHgBHxI7969FRoammc8ISFBy5cvd7vuvHnzlJWVlWe8Q4cOqlq1qtt1ARQ/XKcAeDOCAwDejnspAN6MeykAgDch4AR8SFhYmPr27Ws498Ybb7hVMzs7W9OmTTOcGzRokFs1ARRfXKcAeDOCAwDejnspAN6MeykAgDch4AR8zIgRIwzHV69erYULF7pc75133tHBgwfzjFeuXFkDBgxwuR4AcJ0C4K0IDgD4Au
```

Comment:
Replace this line with ### BEGIN SOLUTION ### (and remove from earlier). The website publishing script with replace code between ### BEGIN SOLUTION ### and ### END SOLUTION ### with # Add your solution here.